pydantic_ai/models/groq.py

```diff
@@ -0,0 +1,372 @@
+from __future__ import annotations as _annotations
+
+from collections.abc import AsyncIterator, Iterable, Mapping, Sequence
+from contextlib import asynccontextmanager
+from dataclasses import dataclass, field
+from datetime import datetime, timezone
+from typing import Literal, overload
+
+from groq import NOT_GIVEN, AsyncGroq, AsyncStream
+from groq.types import chat
+from groq.types.chat import ChatCompletion, ChatCompletionChunk
+from groq.types.chat.chat_completion_chunk import ChoiceDeltaToolCall
```

Comment:
this looks very close to OpenAI's SDK, could we somehow reuse the OpenAI model and just change the endpoint?

Reply:
There are slight differences which is why we need our client (especially in the token use x_groq parts)